test/modules/obsagent-client/src/main.rs

```diff
@@ -120,91 +43,27 @@ fn init_logging() {
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
     init_logging();
-    let config = init_config()?;
+    let config = Arc::new(config::init_config()?);
     info!(
         "Starting Observability Agent Client with configuration: {:?}",
         config
     );
-    let _started = init_meter(1.0 / config.push_rate, config.otlp_endpoint)?;
-    let meter = global::meter(METER_NAME);
 
-    // Init synchronous instruments
-    let counter = meter.u64_counter("u64_counter_example").init();
-    let ud_counter = meter
-        .i64_up_down_counter("i64_up_down_counter_example")
-        .init();
-    let value_recorder = meter
-        .f64_value_recorder("f64_value_recorder_example")
-        .init();
-
-    // Init asynchronous instruments
-    let sum = Arc::new(Mutex::new(0));
-    let sum_clone = Arc::clone(&sum);
-    let sum_observer_cb = move |res: ObserverResult<u64>| {
-        if let Ok(sum_clone) = sum_clone.try_lock() {
-            res.observe(*sum_clone, &[]);
-        } else {
-            error!("try_lock failed")
-        }
-    };
-    let _sum_observer = meter
-        .u64_sum_observer("u64_sum_observer_example", sum_observer_cb)
-        .init();
-    let ud_sum = Arc::new(Mutex::new(0));
-    let ud_sum_clone = Arc::clone(&ud_sum);
-    let ud_sum_observer_cb = move |res: ObserverResult<i64>| {
-        if let Ok(ud_sum_clone) = ud_sum_clone.try_lock() {
-            res.observe(*ud_sum_clone, &[]);
-        } else {
-            error!("try_lock failed")
-        }
-    };
-    let _ud_sum_observer = meter
-        .i64_up_down_sum_observer("i64_up_down_sum_observer_example", ud_sum_observer_cb)
-        .init();
-    let value: Arc<Mutex<f64>> = Arc::new(Mutex::new(0.0));
-    let value_clone = Arc::clone(&value);
-    let value_observer_cb = move |res: ObserverResult<f64>| {
-        if let Ok(value_clone) = value_clone.try_lock() {
-            res.observe(*value_clone, &[]);
-        } else {
-            error!("try_lock failed")
-        }
-    };
-    let _value_observer = meter
-        .f64_value_observer("f64_value_observer_example", value_observer_cb)
-        .init();
+    #[cfg(feature = "otel")]
+    let otel_fut = otel_client::run(config.clone());
+    #[cfg(feature = "prom")]
+    let prom_fut = prometheus_server::run(config.clone());
```

Comment:
There may be a scenario where we don't care about performance and we want to use a single obsagent-client to test the obsagent running in both OTel and Prometheus modes.  This dual mode in the obsagent is very likely to be required so it would be good to have the same in the client.  Also, if we decide to make the modes mutually exclusive, we'll have to reconsider the decision to use Cargo features, because the Cargo documentation recommends [avoiding mutually exclusive features](https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features).

Reply:
> Also, if we decide to make the modes mutually exclusive, we'll have to reconsider the decision to use Cargo features, because the Cargo documentation recommends avoiding mutually exclusive features.

If we go this way, I am fine hacking around that part and just printing a message saying that running with both features is not an intended use case.

I am fine if we keep the dual mode in this test client, but I am a bit wary of the two clients competing for resources that may adversely affect benchmarking. For this reason running the clients in separate containers would make this less of a risk. 

If you are keeping the dual mode I recommend making sure the two clients get spawned earlier onto new threads (i.e. spawning on each client's run method). This could help to avoid some starvation issues.

Reply:
I see you already fixed the spawning. Nice!